### PR TITLE
Bump libraries to solve build errors with supporting oauth

### PR DESCRIPTION
**Pull request by LittleWat**
Sorry, [the previous PR](https://github.com/Apicurio/apicurio-registry-content-sync-operator/pull/135) caused the build error so let me update the libraries to create the image  with supporting oauth

This PR is based on the following PRs:
- [Bump quarkus-operator-sdk from 2.0.3 to 3.0.2 by dependabot[bot] · Pull Request #40 · Apicurio/apicurio-registry-content-sync-operator](https://github.com/Apicurio/apicurio-registry-content-sync-operator/pull/40)
- [Support for OAuth by vsevel · Pull Request #142 · Apicurio/apicurio-registry-content-sync-operator](https://github.com/Apicurio/apicurio-registry-content-sync-operator/pull/142)

Added commits for building the image successfully and making the operator work as expected.

The followings are necessary to be fixed 🤦  (I would be glad if you could share the knowledge on how to fix 🙏 ).

- Make test work
  - Tests are still failing, but docker image build `$ make build-image-faster` has succeeded
- Generate `Pets` bean successfully
  - Pets whose "type": "array" in `sync/src/test/resources/artifactTypes/openapi/3.0-petstore_v1.json` does not produce `Pets` class after upgrading the library :thinking:

### CODE REVIEW

File: sync/src/test/resources/artifactTypes/openapi/3.0-petstore_v1.json [hunk 1]
```diff
@@ -151,7 +151,7 @@
         }
       },
       "Pets": {
-        "type": "array",
+        "type": "object",
```
Comment (LittleWat) [inline sync/src/test/resources/artifactTypes/openapi/3.0-petstore_v1.json:154]: I don't know why but `array` cannot generate `Pets` object.

I got the following error:

```java
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:testCompile (default-testCompile) on project kube-sync-server: Compilation failure: Compilation failure: 
[ERROR] /home/kohei/tmp/apicurio-registry-content-sync-operator/sync/target/generated-test-sources/jaxrs/io/apicurio/registry/rest/v2/PetsResource.java:[4,42] cannot find symbol
[ERROR]   symbol:   class Pets
[ERROR]   location: package io.apicurio.registry.rest.v2.beans
[ERROR] /home/kohei/tmp/apicurio-registry-content-sync-operator/sync/target/generated-test-sources/jaxrs/io/apicurio/registry/rest/v2/PetsResource.java:[19,3] cannot find symbol
[ERROR]   symbol:   class Pets
[ERROR]   location: interface io.apicurio.registry.rest.v2.PetsResource
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:testCompile (default-testCompile) on project kube-sync-server: Compilation failure
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
    at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:1212)
    at org.apache.maven.plugin.compiler.TestCompilerMojo.execute (TestCompilerMojo.java:199)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :kube-sync-server
```

**Comment by LittleWat** [comment]
I updated the `apicurio.registry.version` from 2.4.1.Final to 2.4.3.Final but the error is not changed 🤦 

In  2.4.3.Final, the `apicurio-common-rest-client` version seems to be `0.1.17`
https://github.com/Apicurio/apicurio-registry/blob/0055c06029a0dc34f234f2fa967538354ca384b7/pom.xml#L212

But in this project, , the `apicurio-common-rest-client` version is set  `0.1.15`

```shell
$  mvn dependency:tree  
....
[INFO] 
[INFO] --- dependency:2.8:tree (default-cli) @ kube-sync-api ---
[INFO] io.apicurio:kube-sync-api:jar:1.0.0-SNAPSHOT
[INFO] +- io.apicurio:apicurio-registry-client:jar:2.4.3.Final:compile
[INFO] |  +- io.apicurio:apicurio-registry-common:jar:2.4.3.Final:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.14.2:compile
[INFO] |  +- io.apicurio:apicurio-common-rest-client-jdk:jar:0.1.15.Final:compile
[INFO] |  |  +- io.apicurio:apicurio-common-rest-client-common:jar:0.1.15.Final:compile
[INFO] |  |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.14.2:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-databind:jar:2.14.2:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:1.7.36:compile
...
```

@carlesarnal I'm so sorry to mention you but would you know the reason...? 🙏 Thanks every time 🙇

**Comment by carlesarnal** [comment]
@LittleWat Thanks for trying to add the support. I have merged https://github.com/Apicurio/apicurio-registry-content-sync-operator/pull/145 with the oauth support (sometimes it may be hard to update something without the context). If you would like to contribute, it should be a bit easier now. Thanks!

**Comment by LittleWat** [comment]
@carlesarnal Thank you for fixing and supporting the OAUTH feature!! 🙇‍♂️ It helped our project a lot!

**Pull request closed without merging**